src/sql/src/query_model/hir.rs

```diff
@@ -94,7 +94,14 @@ impl FromHir {
                 }))
             }
             HirRelationExpr::Map { input, mut scalars } => {
-                let mut box_id = self.generate_select(*input);
+                let mut box_id = self.generate_internal(*input);
```

Comment:
Good catch! The code looks perfect!

(BTW There is a plan to be able to generate Hir manually for testing purposes so that we can force any Hir to be fed to this lowering code.)

Reply:
🎉 thanks very much for taking a look!